BlueprintUI/Sources/Layout/KeyboardReader.swift

```diff
@@ -0,0 +1,228 @@
+//
+//  KeyboardReader.swift
+//  BlueprintUI
+//
+//  Created by Kyle Van Essen on 9/30/20.
+//
+
+import Foundation
+
+
+///
+/// A `KeyboardReader` is used to build an element which is responsive to the position of
+/// the on-screen iOS keyboard. This allows you to customize your elements based on the
+/// state of the on-screen keyboard.
+///
+/// While Blueprint's `ScrollView` already adjusts insets for the keyboard, other custom
+/// elements and UIs do not. Use a `KeyboardReader` to implement keyboard management
+/// within your element.
+///
+/// The initializer for `KeyboardReader` takes an escaping closure, which is called each time
+/// the keyboard frame changes, or the element's frame changes, in order to update the element
+/// to account for the new keyboard position.
+///
+/// ```
+/// KeyboardReader { info in
+///     myElement.inset(bottom: info.keyboardFrame.height)
+/// }
+/// ```
+public struct KeyboardReader : ProxyElement {
```

Comment:
Likely also consider some sort of inputAccessoryView integration here? Eg for bottom-pinned buttons or bars, that's the only way to get the interactive dismissal behaviour.

Reply:
This definitely simplifies a lot of common cases